### PR TITLE
train crud

### PR DESCRIPTION
**Pull request by maxwoon-ms**
## Description of work

added 
- CRUD for train api
- request specs for train controller

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Chore (Minor requested change)

## Checklist

- [x] I have added tests to prove my feature is functional
- [x] My code is sufficiently documented, include sample of how to use the component

### CODE REVIEW

**Review by kristianjabilles-ms — APPROVED** [review]
LGTM

File: spec/requests/trains_request_spec.rb [hunk 1]
```diff
@@ -29,4 +49,92 @@
...
+      end
+    end
+  end
+
+  describe 'PUT /v1/trains/1' do
+    context 'with valid attributes' do
+      it 'returns http status 200' do
+        put v1_train_path(train), params: { train: valid_attributes }
+        expect(response).to be_ok
+        expect(JSON.parse(response.body)).to eq({"message"=>"name successfully updated"})
```
Comment (benrobinson-ms) [inline spec/requests/trains_request_spec.rb:102]: assert the train in question was actually updated too please.
we are just asserting that the messages appear as expected, but not that we have update the train record we pass in

File: spec/requests/trains_request_spec.rb [hunk 2]
```diff
@@ -29,4 +49,92 @@
       expect(JSON.parse(response.body)).to eq(expected_response)
     end
   end
+
+  describe 'POST /v1/trains' do
+    context 'with valid attributes' do
+      it 'returns http status 201' do
+        post v1_trains_path params: { train: expected_response }
+        expect(response).to be_created
+        expect(JSON.parse(response.body)).to eq({"message"=>"#{train.name} successfully created"})
```
Comment (benrobinson-ms) [inline spec/requests/trains_request_spec.rb:60]: we should verify a train was actually created.
we are only verifying that the messages and status from the server are correct, but not that we have actually created a new record

File: spec/requests/trains_request_spec.rb [hunk 3]
```diff
@@ -29,4 +49,92 @@
...
+        })
+      end
+    end
+  end
+
+  describe 'DELETE /v1/trains/1' do
+    it 'returns http status 200' do
+      delete v1_train_path(train)
+      expect(response).to be_ok
+      expect(JSON.parse(response.body)).to eq({"message"=>"#{train.name} successfully deleted"})
```
Comment (benrobinson-ms) [inline spec/requests/trains_request_spec.rb:143]: i think we might also be able to verify the product doesnt exist in the system anymore too.
or atleast something was removed from the count

**Review by benrobinson-ms — COMMENTED** [review]
Looks good, just some extra info in the tests imo

File: spec/requests/trains_request_spec.rb (hunk 2 above)
Comment (maxwoon-ms) [reply spec/requests/trains_request_spec.rb:60]: can i use `count` for this?

Comment (benrobinson-ms) [reply spec/requests/trains_request_spec.rb:60]: probably

**Review by benrobinson-ms — APPROVED** [review]

**Pull request merged**